File: javapatterns/src/main/java/org/iut/nantes/strategy/crypto/SHA512.java
package org.iut.nantes.strategy.crypto;

import javax.xml.bind.DatatypeConverter;
import java.security.*;
import java.math.*;

public class SHA512 implements CrypteStrategy
{
  private MessageDigest m;

  public SHA512() throws Exception
  {
    m = MessageDigest.getInstance( "SHA-512" );
  }

  public String encode(String s)
  {
    return DatatypeConverter.printHexBinary( m.digest( s.getBytes() ) );
  }
}

File: src/fr/tblf/patterns/strategy/crypto/comportements/SHA512.kt
package fr.tblf.patterns.strategy.crypto.comportements

import java.security.MessageDigest

class SHA512 : CryptoStrat {
    private val messageDigest: MessageDigest = MessageDigest.getInstance("SHA-512")

    override fun encrypt(message: String) {
        println(String(messageDigest.digest(message.toByteArray())))
    }
}